flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -557,3 +1022,242 @@ def downscale_hmax(self, zsmax, demfile: Path):
             hmin=0.01,
         )
         return hmax
+
+    def _calculate_rp_floodmaps(self):
```

Comment:
I think we should make this more generic and move it outside of the specific model adapter. It probably needs an overhaul anyway because I do not think that it will be able to handle large model domains.

Reply:
Yeah the eventsets are a beast I havent really touched that much because it is so much more than just running a single event. It might even be nice to move this probabilistic part away from the event class and into its own distinct class/tab.

Reply:
I like that. And ideally we move the return period flood map creation into its own package because other projects want to use that, too. But not necessarily now.